diff --git a/src/core/SkTypeface.cpp b/src/core/SkTypeface.cpp
index cc0bd3f025..b8d469abf1 100644
--- a/src/core/SkTypeface.cpp
+++ b/src/core/SkTypeface.cpp
@@ -310,7 +310,11 @@ std::unique_ptr<SkAdvancedTypefaceMetrics> SkTypeface::getAdvancedMetrics() cons
     if (result && result->fPostScriptName.isEmpty()) {
         result->fPostScriptName = result->fFontName;
     }
-    if (result && result->fType == SkAdvancedTypefaceMetrics::kTrueType_Font) {
+    if (result && result->fType == SkAdvancedTypefaceMetrics::kOther_Font) {
+        result->fType = SkAdvancedTypefaceMetrics::kCFF_Font;
+    }
+    if (result && (result->fType == SkAdvancedTypefaceMetrics::kCFF_Font ||
+                   result->fType == SkAdvancedTypefaceMetrics::kTrueType_Font)) {
         SkOTTableOS2::Version::V2::Type::Field fsType;
         constexpr SkFontTableTag os2Tag = SkTEndian_SwapBE32(SkOTTableOS2::TAG);
         constexpr size_t fsTypeOffset = offsetof(SkOTTableOS2::Version::V2, fsType);
diff --git a/src/pdf/SkPDFFont.cpp b/src/pdf/SkPDFFont.cpp
index e9976506d5..2afeb8b746 100644
--- a/src/pdf/SkPDFFont.cpp
+++ b/src/pdf/SkPDFFont.cpp
@@ -254,6 +254,7 @@ sk_sp<SkPDFFont> SkPDFFont::GetFontResource(SkPDFCanon* canon,
     sk_sp<SkPDFFont> font;
     switch (type) {
         case SkAdvancedTypefaceMetrics::kType1CID_Font:
+        case SkAdvancedTypefaceMetrics::kCFF_Font:
         case SkAdvancedTypefaceMetrics::kTrueType_Font:
             SkASSERT(multibyte);
             font = sk_make_sp<SkPDFType0Font>(std::move(info), metrics);
@@ -439,6 +440,7 @@ void SkPDFType0Font::getFontSubset(SkPDFCanon* canon) {
                 descriptor->insertObjRef("FontFile2", std::move(fontStream));
                 break;
             }
+            case SkAdvancedTypefaceMetrics::kCFF_Font:
             case SkAdvancedTypefaceMetrics::kType1CID_Font: {
                 auto fontStream = sk_make_sp<SkPDFSharedStream>(std::move(fontAsset));
                 fontStream->dict()->insertName("Subtype", "CIDFontType0C");
@@ -456,6 +458,7 @@ void SkPDFType0Font::getFontSubset(SkPDFCanon* canon) {
 
     switch (type) {
         case SkAdvancedTypefaceMetrics::kType1CID_Font:
+        case SkAdvancedTypefaceMetrics::kCFF_Font:
             newCIDFont->insertName("Subtype", "CIDFontType0");
             break;
         case SkAdvancedTypefaceMetrics::kTrueType_Font:
diff --git a/src/pdf/SkPDFFont.h b/src/pdf/SkPDFFont.h
index 1441eedad9..9f189203f4 100644
--- a/src/pdf/SkPDFFont.h
+++ b/src/pdf/SkPDFFont.h
@@ -46,6 +46,7 @@ public:
 
     static bool IsMultiByte(SkAdvancedTypefaceMetrics::FontType type) {
         return type == SkAdvancedTypefaceMetrics::kType1CID_Font ||
+               type == SkAdvancedTypefaceMetrics::kCFF_Font ||
                type == SkAdvancedTypefaceMetrics::kTrueType_Font;
     }
 
diff --git a/src/ports/SkFontHost_mac.cpp b/src/ports/SkFontHost_mac.cpp
index 18d09b4ec0..29902dd132 100644
--- a/src/ports/SkFontHost_mac.cpp
+++ b/src/ports/SkFontHost_mac.cpp
@@ -1808,7 +1808,7 @@ static SK_SFNT_ULONG get_font_type_tag(CTFontRef ctFont) {
         case kCTFontFormatTrueType:
             return SkSFNTHeader::fontType_MacTrueType::TAG;
         case kCTFontFormatPostScript:
-            return SkSFNTHeader::fontType_PostScript::TAG;
+            return SkSFNTHeader::fontType_OpenTypeCFF::TAG; // was fontType_PostScript::TAG ('typ1')
         case kCTFontFormatBitmap:
             return SkSFNTHeader::fontType_MacTrueType::TAG;
         case kCTFontFormatUnrecognized:
