diff --git a/src/core/SkTypeface.cpp b/src/core/SkTypeface.cpp
index 3e485f5ec1..40ed89cb94 100644
--- a/src/core/SkTypeface.cpp
+++ b/src/core/SkTypeface.cpp
@@ -344,7 +344,11 @@ std::unique_ptr<SkAdvancedTypefaceMetrics> SkTypeface::getAdvancedMetrics() cons
     if (result && result->fPostScriptName.isEmpty()) {
         result->fPostScriptName = result->fFontName;
     }
-    if (result && result->fType == SkAdvancedTypefaceMetrics::kTrueType_Font) {
+    if (result && result->fType == SkAdvancedTypefaceMetrics::kOther_Font) {
+        result->fType = SkAdvancedTypefaceMetrics::kCFF_Font;
+    }
+    if (result && (result->fType == SkAdvancedTypefaceMetrics::kCFF_Font ||
+                   result->fType == SkAdvancedTypefaceMetrics::kTrueType_Font)) {
         SkOTTableOS2::Version::V2::Type::Field fsType;
         constexpr SkFontTableTag os2Tag = SkTEndian_SwapBE32(SkOTTableOS2::TAG);
         constexpr size_t fsTypeOffset = offsetof(SkOTTableOS2::Version::V2, fsType);
diff --git a/src/pdf/SkPDFFont.cpp b/src/pdf/SkPDFFont.cpp
index d3f3457e80..e0cd9a58e4 100644
--- a/src/pdf/SkPDFFont.cpp
+++ b/src/pdf/SkPDFFont.cpp
@@ -338,6 +338,7 @@ static void emit_subset_type0(const SkPDFFont& font, SkPDFDocument* doc) {
                 descriptor->insertRef("FontFile2", doc->serialize(fontStream));
                 break;
             }
+            case SkAdvancedTypefaceMetrics::kCFF_Font:
             case SkAdvancedTypefaceMetrics::kType1CID_Font: {
                 auto fontStream = sk_make_sp<SkPDFSharedStream>(std::move(fontAsset));
                 fontStream->dict()->insertName("Subtype", "CIDFontType0C");
@@ -355,6 +356,7 @@ static void emit_subset_type0(const SkPDFFont& font, SkPDFDocument* doc) {
 
     switch (type) {
         case SkAdvancedTypefaceMetrics::kType1CID_Font:
+        case SkAdvancedTypefaceMetrics::kCFF_Font:
             newCIDFont->insertName("Subtype", "CIDFontType0");
             break;
         case SkAdvancedTypefaceMetrics::kTrueType_Font:
@@ -769,6 +771,7 @@ void SkPDFFont::emitSubset(SkPDFDocument* doc) const {
     SkASSERT(fFontType != SkPDFFont().fFontType); // not default value
     switch (fFontType) {
         case SkAdvancedTypefaceMetrics::kType1CID_Font:
+        case SkAdvancedTypefaceMetrics::kCFF_Font:
         case SkAdvancedTypefaceMetrics::kTrueType_Font:
             return emit_subset_type0(*this, doc);
         case SkAdvancedTypefaceMetrics::kType1_Font:
diff --git a/src/pdf/SkPDFFont.h b/src/pdf/SkPDFFont.h
index 4d3ee9873f..7accd93308 100644
--- a/src/pdf/SkPDFFont.h
+++ b/src/pdf/SkPDFFont.h
@@ -43,6 +43,7 @@ public:
 
     static bool IsMultiByte(SkAdvancedTypefaceMetrics::FontType type) {
         return type == SkAdvancedTypefaceMetrics::kType1CID_Font ||
+               type == SkAdvancedTypefaceMetrics::kCFF_Font ||
                type == SkAdvancedTypefaceMetrics::kTrueType_Font;
     }
 
diff --git a/src/ports/SkFontHost_mac.cpp b/src/ports/SkFontHost_mac.cpp
index d31916d306..4312c15965 100644
--- a/src/ports/SkFontHost_mac.cpp
+++ b/src/ports/SkFontHost_mac.cpp
@@ -1773,7 +1773,7 @@ static SK_SFNT_ULONG get_font_type_tag(CTFontRef ctFont) {
         case kCTFontFormatTrueType:
             return SkSFNTHeader::fontType_MacTrueType::TAG;
         case kCTFontFormatPostScript:
-            return SkSFNTHeader::fontType_PostScript::TAG;
+            return SkSFNTHeader::fontType_OpenTypeCFF::TAG; // was fontType_PostScript::TAG ('typ1')
         case kCTFontFormatBitmap:
             return SkSFNTHeader::fontType_MacTrueType::TAG;
         case kCTFontFormatUnrecognized:
